src/addPropTrackingRef.ts

```diff
@@ -0,0 +1,29 @@
+import {CurriedPropsAdder, flowMax, addRef, addEffect} from 'ad-hok'
+import {MutableRefObject} from 'react'
+
+type AddPropTrackingRefType = <
+  TProps,
+  TPropName extends keyof TProps,
+  TRefPropName extends string
+>(
+  propName: TPropName,
+  refPropName: TRefPropName,
+) => CurriedPropsAdder<
+  TProps,
+  {
+    [refPropName in TRefPropName]: MutableRefObject<TProps[TPropName]>
+  }
+>
+
+const addPropTrackingRef: AddPropTrackingRefType = (propName, refPropName) =>
+  flowMax(
+    addRef(refPropName, null as any),
```

Comment:
Should probably just initialize the ref with the current/initial value of the `propName` prop, that seems correct behaviorally and would allow strongly typing the ref I think